src/plugins/plugin-http.js

```diff
@@ -76,8 +76,22 @@ function patchRequest (http, api) {
       var req = request.call(this, options, function(res) {
         api.wrapEmitter(res);
         var numBytes = 0;
-        res.on('data', function (chunk) {
-          numBytes += chunk.length;
+        var listenerAttached = false;
```

Comment:
Would this mechanism still work if the use calls `res.read()` rather than `res.on('data')` to consume the data?